modules/client/src/internal/client/methods.ts

```diff
@@ -625,4 +637,62 @@ export class ClientMethods extends ClientCore implements IClientMethods {
       ),
     );
   }
+
+  public async getPlugins({
+    limit = 10,
+    skip = 0,
+    direction = SortDirection.ASC,
+    sortBy = PluginSortBy.SUBDOMAIN,
+    subdomain,
+  }: PluginQueryParams): Promise<PluginRepo[]> {
```

Comment:
Like this, I think we are expecting `{}` even is no params are passed

```suggestion
  public async getPlugins({
    limit = 10,
    skip = 0,
    direction = SortDirection.ASC,
    sortBy = PluginSortBy.SUBDOMAIN,
    subdomain,
  }: PluginQueryParams = {}): Promise<PluginRepo[]> {
```

Reply:
Yes but this does not solve the issue of needing to call it with `{}`. One solution would be having an interface with the optional parameter that calls the function with the mandatory parameter:

```ts
interface Test {
  someFunction: (params?: SomeType) => Promise<SomeTypeResponse>;
}

class Methods implements Test {
  someFunction(params?: SomeType): Promise<SomeTypeResponse> {
    return this.someFunctionImplementation(params || {});
  }

  private someFunctionImplementation(
    { prop1 = "ss", prop2 = "bb" }: SomeType,
  ): Promise<SomeTypeResponse> {
    // DO Stuff
  }
}
```

Reply:
It does solve the problem. If the parameter has a default value `= {}` then you don't need to pass anything. If you don't, `{}` will be used and all parameters are going to be empty or have the default value

Reply:
Typescript complains, I tested it. Maybe its because this should be done in the interface?